app/uk/gov/hmrc/apinotificationqueue/repository/NotificationRepository.scala

```diff
@@ -105,24 +105,37 @@ class NotificationMongoRepository @Inject()(mongoDbProvider: MongoDbProvider,
 
     val clientNotification = ClientNotification(clientId, notification)
 
-    lazy val errorMsg = s"Notification not updated for client $clientId, notificationId: ${notification.notificationId}"
+    lazy val errorMsg = s"Notification not updated for clientId $clientId, notificationId: ${notification.notificationId}"
 
     val selector = Json.obj("clientId" -> clientId, "notification.notificationId" -> notification.notificationId)
 
-    collection.update(selector, clientNotification).map {
-      writeResult => notificationRepositoryErrorHandler.handleSaveError(writeResult, errorMsg, notification)
+    findAndUpdate(selector, ClientNotificationJF.writes(clientNotification)).map {
```

Comment:
would it make sense to extract the new error handling code to the existing `NotificationRepositoryErrorHandler` ?